src/operator/nn/mkldnn/mkldnn_act.cc

```diff
@@ -135,24 +135,21 @@ static MKLDNNActForward &GetActForward(const ActivationParam& param,
                                        const OpContext &ctx, const NDArray &in_data,
                                        const mkldnn::memory &in_mem) {
 #if DMLC_CXX11_THREAD_LOCAL
-  static thread_local std::unordered_map<MKLDNNActSignature, MKLDNNActForward, OpHash> fwds;
+  static thread_local MKLDNNCache<MKLDNNActSignature, MKLDNNActForward, OpHash> fwds;
```

Comment:
one of the reason why the memory usage is high could be because it maintains a cache for each thread? 
I am wondering if the original authors measured contention and performance impact if we maintained one cache for the process? 
I understand it *might* need some kind of locking, has someone explored non-blocking/lockfree data structures?